### ISSUE TITLE
After activating "Show Desktop" and after double-clicking a file on Desktop, Show Desktop doesn't restore windows

### ISSUE THREAD

**Issue by nick-s-b** [linuxmint/cinnamon#5776]
I'm using the latest Cinnamon with LM18: `Desktop: Cinnamon 3.0.7`

Steps to reproduce:
0a) For convenience, assign Show Desktop to one of the Hot Corners
0b) Place an image or some other file on the Desktop so you have something to click
0c) Have few application windows open
1) Activate "Show Desktop"
2) Double-click on a file/image on the desktop
3) Your image viewer will open but you previous windows will not be restored and will become minimized
4) Activating Show Desktop again in an attempt to show the windows will not bring them back. They will remain minimized.

This bug greatly reduces the usability of Show Desktop feature.
Thanks!

**Comment by JosephMcc** [comment]
I can reproduce this. Not sure how this should work though. The current behavior is a convenient way to minimize all windows and open a new one.

Do you expect when double clicking to open a new window, everything should be restored?

**Comment by nick-s-b** [comment]
> Do you expect when double clicking to open a new window, everything should be restored?

Yeah, everything should be restored. This whole concept is akin to old Expose from OSX. If you try that same sequence in OSX, it will restore all the windows. That actually makes perfect sense too.

Say you're working on something and you have a bunch of files on the Desktop (not the neatest thing to do but it's convenient) and you want to open a file. You activate Show Desktop, you find the file, you double click it to open it, and everything else is restored. The current behavior makes little sense and doesn't really fit in the workflow. Show Desktop's main feature shouldn't be to minimize all windows. It should then be renamed to that instead if that's what people really want.

**Comment by hyOzd** [comment]
> Yeah, everything should be restored.

That's not the behavior I would expect. Some people click on "Show Desktop" to clear their screen. And focus on the newly opened window. Especially those who don't use multiple workspaces.

**Comment by nick-s-b** [comment]
>That's not the behavior I would expect. Some people click on "Show Desktop" to clear their screen. And focus on the newly opened window. Especially those who don't use multiple workspaces.

Then why does clicking it again bring everything back? And why would just clicking on an icon on the desktop invalidate the windows positions and minimize all windows and prevent them from being restored? That feature should then be renamed to "Minimize all windows" and not "Show Desktop"..

**Comment by hyOzd** [comment]
> Then why does clicking it again bring everything back?

A reason I like this behavior is; it allows me to quickly access desklets. I press Super+D see the desklet, press Super+D again to return back to my full screen window. To be honest they are not that popular.

> And why would just clicking on an icon on the desktop invalidate the windows positions and minimize all windows and prevent them from being restored?

As I said, to focus on the newly opened window without distraction from others.

> That feature should then be renamed to "Minimize all windows" and not "Show Desktop"..

Not a bad idea. Maybe we can have both as separate applets?

**Comment by nick-s-b** [comment]
This feature is most useful for people with small screens (i.e. laptops). I use it on my old ThinkPad with a small screen and it's incredibly frustrating that it doesn't behave as it behaves on other OSes/DEs. If you have a small screen, one of the most convenient places to place your folders/documents/symlinks is on the desktop. But due to screen-space requirements, most of the time they will be invisible to you. So if you want to access them, "Show Desktop" quickly reveals them. But if you do **anything** after activating Show Desktop (except activating it back again), you will lose your windows. That has very little usability.

**Comment by Vahan86** [comment]
+1 for # 4. In my opinion after opening the new window, activating "show desktop" again should restore previously opened windows as well.

But I am not sure that "double clicking to open a new window should restore minimized windows as well" behavior is better than existing one.
